integration-tests/tests/bombastic.rs

```diff
@@ -1,126 +1,194 @@
-use integration_tests::{assert_within_timeout, with_bombastic};
+use integration_tests::{assert_within_timeout, BombasticContext, ProviderContext};
 use reqwest::StatusCode;
 use serde_json::{json, Value};
 use std::time::Duration;
+use test_context::test_context;
+use trustification_auth::client::TokenInjector;
 use urlencoding::encode;
 
-#[test]
-fn test_upload() {
-    with_bombastic(Duration::from_secs(60), |port| async move {
-        let input = serde_json::from_str(include_str!("../../bombastic/testdata/my-sbom.json")).unwrap();
-        let id = "test-upload";
-        upload(port, id, &input).await;
-        let response = reqwest::get(format!("http://localhost:{port}/api/v1/sbom?id={id}"))
-            .await
-            .unwrap();
-        assert_eq!(response.status(), StatusCode::OK);
-        let output: Value = response.json().await.unwrap();
-        assert_eq!(input, output);
-    })
+#[test_context(BombasticContext)]
+#[tokio::test]
+#[ntest::timeout(60_000)]
+async fn test_upload(context: &mut BombasticContext) {
+    let input = serde_json::from_str(include_str!("../../bombastic/testdata/my-sbom.json")).unwrap();
+    let id = "test-upload";
+    upload(context.port, id, &input, &context.provider).await;
+    let response = reqwest::Client::new()
+        .get(format!(
+            "http://localhost:{port}/api/v1/sbom?id={id}",
+            port = context.port
+        ))
+        .inject_token(&context.provider.provider_manager)
+        .await
+        .unwrap()
```

Comment:
Does this absolutely have to be async and return an Error? Really adds a lot of noise to the tests.